vendor/k8s.io/kubernetes/pkg/kubectl/cmd/get.go

```diff
@@ -328,9 +328,7 @@ func RunGet(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args [
 		}
 
 		var obj runtime.Object
-		if singular {
-			obj = infos[0].Object
-		} else {
+		if !singular || len(infos) > 1 {
```

Comment:
There's an upstream test-cmd test for it. But the PR hasn't merged upstream, and I had turned it over to AdoHe to finish it, so this commit is potentially getting replaced.